src/main/java/org/openhab/binding/zwave/ZWaveBindingConstants.java

```diff
@@ -89,14 +92,60 @@
-    public final static String OFFLINE_CONTROLLER_OFFLINE_DEFAULT = "Controller is offline";
-    public final static String OFFLINE_NODE_DEAD_KEY = "zwave.thingstate.node_dead";
-    public final static String OFFLINE_NODE_DEAD_DEFAULT = "Node is not communicating with controller";
-    public final static String OFFLINE_NODE_NOTFOUND_KEY = "zwave.thingstate.node_notfound";
-    public final static String OFFLINE_NODE_NOTFOUND_DEFAULT = "Node not found in Z-Wave network";
+    public final static I18nConstant OFFLINE_CTLR_OFFLINE = new I18nConstant("zwave.thingstate.controller_offline",
+            "Controller is offline");
+    public final static I18nConstant OFFLINE_NODE_DEAD = new I18nConstant("zwave.thingstate.node_dead",
+            "Node is not communicating with controller");
+    public final static I18nConstant OFFLINE_NODE_NOTFOUND = new I18nConstant("zwave.thingstate.node_notfound",
+            "Node not found in Z-Wave network");
+    public final static I18nConstant OFFLINE_SERIAL_EXISTS = new I18nConstant("zwave.thingstate.serial_notfound",
+            "Serial Error: Port {0} does not exist");
+    public final static I18nConstant OFFLINE_SERIAL_INUSE = new I18nConstant("zwave.thingstate.serial_inuse",
+            "Serial Error: Port {0} is in use");
+    public final static I18nConstant OFFLINE_SERIAL_UNSUPPORTED = new I18nConstant(
+            "zwave.thingstate.serial_unsupported", "Serial Error: Unsupported operation on port {0}");
+    public final static I18nConstant OFFLINE_SERIAL_LISTENERS = new I18nConstant("zwave.thingstate.serial_listeners",
+            "Serial Error: Too many listeners on port {0}");
 
     public final static Integer ACTION_CHECK_VALUE = new Integer(-232323);
 
     public final static Set<ThingTypeUID> SUPPORTED_BRIDGE_TYPES_UIDS = ImmutableSet.of(CONTROLLER_SERIAL);
+
+    private static I18nProvider i18nProvider;
+
+    protected void setI18nProvider(I18nProvider i18nProvider) {
+        ZWaveBindingConstants.i18nProvider = i18nProvider;
+    }
+
+    protected void unsetI18nProvider(I18nProvider i18nProvider) {
+        ZWaveBindingConstants.i18nProvider = null;
+    }
+
+    public static String getI18nConstant(I18nConstant constant) {
+        I18nProvider i18nProviderLocal = i18nProvider;
+        if (i18nProviderLocal == null) {
+            return MessageFormat.format(constant.key, (Object[]) null);
+        }
+        return i18nProvider.getText(ZWaveActivator.getContext().getBundle(), constant.key, constant.defaultText, null,
+                (Object[]) null);
+    }
+
+    public static String getI18nConstant(I18nConstant constant, Object... arguments) {
+        I18nProvider i18nProviderLocal = i18nProvider;
+        if (i18nProviderLocal == null) {
+            return MessageFormat.format(constant.key, arguments);
+        }
+        return i18nProvider.getText(ZWaveActivator.getContext().getBundle(), constant.key, constant.defaultText, null,
```

Comment:
Ah - damn - well spotted :)

> On 21 Aug 2016, at 14:05, Markus Rathgeb <notifications@github.com> wrote:
> 
> In src/main/java/org/openhab/binding/zwave/ZWaveBindingConstants.java <https://github.com/openhab/org.openhab.binding.zwave/pull/40#discussion_r75594905>:
> 
> > +
> > +    public static String getI18nConstant(I18nConstant constant) {
> > +        I18nProvider i18nProviderLocal = i18nProvider;
> > +        if (i18nProviderLocal == null) {
> > +            return MessageFormat.format(constant.key, (Object[]) null);
> > +        }
> > +        return i18nProvider.getText(ZWaveActivator.getContext().getBundle(), constant.key, constant.defaultText, null,
> > +                (Object[]) null);
> > +    }
> > +
> > +    public static String getI18nConstant(I18nConstant constant, Object... arguments) {
> > +        I18nProvider i18nProviderLocal = i18nProvider;
> > +        if (i18nProviderLocal == null) {
> > +            return MessageFormat.format(constant.key, arguments);
> > +        }
> > +        return i18nProvider.getText(ZWaveActivator.getContext().getBundle(), constant.key, constant.defaultText, null,
> I assume you want to use the local copied reference 18nProviderLocal here
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/openhab/org.openhab.binding.zwave/pull/40/files/ff92142cc5c91cd9cd9be52b9df9f06b94debd55#r75594905>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA_kQxxx1jkqMXQxvhQb_ULf5oGym0MPks5qiE0kgaJpZM4JpI6Y>.
>